### ISSUE TITLE
OS X 10.14 Mojave GUI Crashes

### ISSUE THREAD

**Issue by EVWang** [taichi-dev/taichi#355]
**Describe the bug**
run 99line code with error
environment： Mac os Mojave 10.14；python3.7  taichi 0.3.20

**Screenshots**
Please post the full log of the program (instead of just a few lines around the error message). This will help us diagnose what's happening.

**To Reproduce**
```
[T 01/08/20 15:02:18.542] [logging.cpp:Logger@68] Taichi core started. Thread ID = 10734
[Taichi version 0.3.20, cpu only, commit 1c85d8e1]
[W 01/08/20 15:02:18.556] [program.cpp:Program@158] Taichi is not compiled with CUDA.
[W 01/08/20 15:02:18.556] [program.cpp:Program@159] Falling back to x86_64
[I 01/08/20 15:02:18.581] [taichi_llvm_context.cpp:TaichiLLVMContext@59] Creating llvm context for arch: x86_64
[I 01/08/20 15:02:18.602] [/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/taichi/lang/impl.py:materialize@125] Materializing layout...
[D 01/08/20 15:02:18.602] [snode.cpp:create_node@48] Non-power-of-two node size 9000 promoted to 16384.
[D 01/08/20 15:02:18.603] [snode.cpp:create_node@48] Non-power-of-two node size 9000 promoted to 16384.
[D 01/08/20 15:02:18.603] [snode.cpp:create_node@48] Non-power-of-two node size 9000 promoted to 16384.
[D 01/08/20 15:02:18.603] [snode.cpp:create_node@48] Non-power-of-two node size 9000 promoted to 16384.
[D 01/08/20 15:02:18.603] [snode.cpp:create_node@48] Non-power-of-two node size 9000 promoted to 16384.
[D 01/08/20 15:02:18.603] [snode.cpp:create_node@48] Non-power-of-two node size 9000 promoted to 16384.
[I 01/08/20 15:02:18.751] [struct_llvm.cpp:operator()@276] Allocating data structure of size 1114112 B
Initializing runtime with 26 elements
Runtime initialized.
[E 01/08/20 15:02:19.834] Received signal 11 (Segmentation fault: 11)

                            * Taichi Core - Stack Traceback *                             
==========================================================================================
|                       Module |  Offset | Function                                      |
|----------------------------------------------------------------------------------------|
*               taichi_core.so |     295 | taichi::signal_handler(int)                   |
*     libsystem_platform.dylib |      29 | (null)                                        |
*                          ??? |       0 | (null)                                        |
*                       AppKit |      66 | (null)                                        |
*                       AppKit |       0 | (null)                                        |
*                       AppKit |       0 | (null)                                        |
*                       AppKit |       0 | (null)                                        |
*                       AppKit |       0 | (null)                                        |
*                       AppKit |       0 | (null)                                        |
*                       AppKit |       0 | (null)                                        |
*                       AppKit |       0 | (null)                                        |
*                       AppKit |       0 | (null)                                        |
*                       AppKit |       0 | (null)                                        |
*                       AppKit |       0 | (null)                                        |
*                       AppKit |       0 | (null)                                        |
*                       AppKit |       0 | (null)                                        |
*                       AppKit |       0 | (null)                                        |
*                       AppKit |       0 | (null)                                        |
*               taichi_core.so |     746 | taichi::GUI::create_window()                  |
*               taichi_core.so |     296 | taichi::GUI::GUI(std::__1::basic_string<char, |
                                         |  std::__1::char_traits<char>, std::__1::alloc |
                                         | ator<char> > const&, int, int, bool)          |
*               taichi_core.so |      99 | pybind11::class_<taichi::GUI> pybind11::detai |
                                         | l::argument_loader<pybind11::detail::value_an |
                                         | d_holder&, std::__1::basic_string<char, std:: |
                                         | __1::char_traits<char>, std::__1::allocator<c |
                                         | har> >, taichi::VectorND<2, int, (taichi::Ins |
                                         | tSetExt)0> >::call_impl<void, void pybind11:: |
                                         | detail::initimpl::constructor<std::__1::basic |
                                         | _string<char, std::__1::char_traits<char>, st |
                                         | d::__1::allocator<char> >, taichi::VectorND<2 |
                                         | , int, (taichi::InstSetExt)0> >::execute<pybi |
                                         | nd11::class_<taichi::GUI>, 0>(pybind11::class |
                                         | _<taichi::GUI>&)::'lambda'(pybind11::detail:: |
                                         | value_and_holder&, std::__1::basic_string<cha |
                                         | r, std::__1::char_traits<char>, std::__1::all |
                                         | ocator<char> >, taichi::VectorND<2, int, (tai |
                                         | chi::InstSetExt)0>)&, 0ul, 1ul, 2ul, pybind11 |
                                         | ::detail::void_type>(void pybind11::detail::i |
                                         | nitimpl::constructor<std::__1::basic_string<c |
                                         | har, std::__1::char_traits<char>, std::__1::a |
                                         | llocator<char> >, taichi::VectorND<2, int, (t |
                                         | aichi::InstSetExt)0> >::execute<pybind11::cla |
                                         | ss_<taichi::GUI>, 0>(pybind11::class_<taichi: |
                                         | :GUI>&)::'lambda'(pybind11::detail::value_and |
                                         | _holder&, std::__1::basic_string<char, std::_ |
                                         | _1::char_traits<char>, std::__1::allocator<ch |
                                         | ar> >, taichi::VectorND<2, int, (taichi::Inst |
                                         | SetExt)0>)&&&, std::__1::integer_sequence<uns |
                                         | igned long, 0ul, 1ul, 2ul>, pybind11::detail: |
                                         | :void_type&&)                                 |
*               taichi_core.so |     153 | void pybind11::cpp_function::initialize<void  |
                                         | pybind11::detail::initimpl::constructor<std:: |
                                         | __1::basic_string<char, std::__1::char_traits |
                                         | <char>, std::__1::allocator<char> >, taichi:: |
                                         | VectorND<2, int, (taichi::InstSetExt)0> >::ex |
                                         | ecute<pybind11::class_<taichi::GUI>, 0>(pybin |
                                         | d11::class_<taichi::GUI>&)::'lambda'(pybind11 |
                                         | ::detail::value_and_holder&, std::__1::basic_ |
                                         | string<char, std::__1::char_traits<char>, std |
                                         | ::__1::allocator<char> >, taichi::VectorND<2, |
                                         |  int, (taichi::InstSetExt)0>), void, pybind11 |
                                         | ::detail::value_and_holder&, std::__1::basic_ |
                                         | string<char, std::__1::char_traits<char>, std |
                                         | ::__1::allocator<char> >, taichi::VectorND<2, |
                                         |  int, (taichi::InstSetExt)0>, pybind11::name, |
                                         |  pybind11::is_method, pybind11::sibling, pybi |
                                         | nd11::detail::is_new_style_constructor>(pybin |
                                         | d11::class_<taichi::GUI>&&, void (*)(pybind11 |
                                         | ::detail::value_and_holder&, std::__1::basic_ |
                                         | string<char, std::__1::char_traits<char>, std |
                                         | ::__1::allocator<char> >, taichi::VectorND<2, |
                                         |  int, (taichi::InstSetExt)0>), pybind11::name |
                                         |  const&, pybind11::is_method const&, pybind11 |
                                         | ::sibling const&, pybind11::detail::is_new_st |
                                         | yle_constructor const&)::'lambda'(pybind11::d |
                                         | etail::function_call&)::operator()(pybind11:: |
                                         | detail::function_call&) const                 |
*               taichi_core.so |    2874 | pybind11::cpp_function::dispatcher(_object*,  |
                                         | _object*, _object*)                           |
*                       Python |     206 | (null)                                        |
*                       Python |     227 | (null)                                        |
*                       Python |     143 | (null)                                        |
*                       Python |     135 | (null)                                        |
*                       Python |     201 | (null)                                        |
*                       Python |     297 | (null)                                        |
*                       Python |     433 | (null)                                        |
*                       Python |     420 | (null)                                        |
*                       Python |   25181 | (null)                                        |
*                       Python |    2467 | (null)                                        |
*                       Python |     523 | (null)                                        |
*                       Python |     143 | (null)                                        |
*                       Python |     145 | (null)                                        |
*                       Python |     297 | (null)                                        |
*                       Python |     433 | (null)                                        |
*                       Python |     420 | (null)                                        |
*                       Python |   25508 | (null)                                        |
*                       Python |    2467 | (null)                                        |
*                       Python |     100 | (null)                                        |
*                       Python |     209 | (null)                                        |
*                       Python |     890 | (null)                                        |
*                       Python |    6915 | (null)                                        |
*                       Python |      58 | (null)                                        |
*                libdyld.dylib |       1 | (null)                                        |
*                          ??? |       2 | (null)                                        |
==========================================================================================
```

**Comment by yuanming-hu** [comment]
Hi @EVWang 

Thanks for reporting. I'm on OS X 10.14.2 yet I couldn't reproduce this issue. Could you provide more info about your OS? Is that 10.14.2 or 10.14.3?

**Comment by BigDeviltjj** [comment]
I encountered the same problem on my mac, My mac version is macOS Mojave 10.14.4, python version is 3.7.4 and installed taichi by `python3 -m pip install taichi-nightly`
Also, I tried to run C++ version code of MPM, the program ran normally(images could be saved to disk correctly) except that no GUI was displayed.

**Comment by yuanming-hu** [comment]
Hi guys, we have released v0.3.21 for OS X (thanks to @k-ye), which might have fixed this issue. However, we cannot test on every OS X version. Could you update and let us know if this issue is resolved?

**Comment by BigDeviltjj** [comment]
Hi @yuanming-hu , now taichi can work correctly for me, thank you!
